framework/src/auxkernels/FunctorElementalGradientAux.C

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorElementalGradientAux.h"
+#include "metaphysicl/raw_type.h"
+
+registerMooseObject("MooseApp", FunctorElementalGradientAux);
+registerMooseObject("MooseApp", ADFunctorElementalGradientAux);
+
+template <bool is_ad>
+InputParameters
+FunctorElementalGradientAuxTempl<is_ad>::validParams()
+{
+  InputParameters params = VectorAuxKernel::validParams();
+  params.addClassDescription(
+      "Evaluates the gradient of a functor (variable, function or functor material property) on "
+      "the current element or quadrature point.");
+  params.addRequiredParam<MooseFunctorName>("functor", "The functor to evaluate");
+  params.addParam<MooseFunctorName>("factor", 1, "A factor to apply on the functor");
+  params.addParam<MaterialPropertyName>(
+      "factor_matprop", "A (regular) material property factor to apply on the functor");
```

Comment:
I *think* you should also be able to do a default value of 1 here as well like you do for the functor so that you don't have to do ternary conditions everywhere

Reply:
yeah just tried and it works. I didnt know matprops could be substituted like this.
I guess everything can! functor, functions, variables, matprops.